pkg/apis/core/types.go

```diff
@@ -1503,6 +1506,37 @@ type LocalVolumeSource struct {
 	Path string
 }
 
+// Represents storage that is managed by an external CSI volume driver
+type CSIPersistentVolumeSource struct {
+	// Driver is the name of the driver to use for this volume.
+	// Required.
+	Driver string
+
+	// VolumeHandle is the unique volume name returned by the CSI volume
+	// plugin’s CreateVolume to refer to the volume on all subsequent calls.
+	// Required.
+	VolumeHandle string
+
+	// Optional: MountSecretRef is a reference to the secret object containing
+	// sensitive information to pass to the CSI driver during NodePublish.
+	// This may be empty if no secret is required. If the secret object contains
+	// more than one secret, all secrets are passed.
+	// +optional
+	MountSecretRef *SecretReference
```

Comment:
Secrets were postponed to future release, please remove them.

Reply:
Hmm, github still shows that MountSecretRef is present in the first commit and at the same time it shows this review remark has been resolved / relevant code has been modified.

Reply:
Oh, I see, you put it into the commit with generated code. Sorry for the noise.

Still, IMO it should be squashed into the first commit for further review. Nobody looks at the generated stuff.

Reply:
@jsafrane ah ok, that commit slipped in to the generated stuff.  I will fix that.  Thanks.